<a href="https://colab.research.google.com/github/a1iceee/Deep_Learning_School_1semestr/blob/master/Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Deep Learning (семестр 1, весна 2020): продвинутый поток**
************
##*Homework № 1: Pandas*

In [6]:
import pandas as pd
data = pd.read_csv('/content/drive/My Drive/BlackFriday.csv')
data.head()

User_ID Product_ID Gender  ... Product_Category_2  Product_Category_3 Purchase
0  1000001  P00069042      F  ...                NaN                 NaN     8370
1  1000001  P00248942      F  ...                6.0                14.0    15200
2  1000001  P00087842      F  ...                NaN                 NaN     1422
3  1000001  P00085442      F  ...               14.0                 NaN     1057
4  1000002  P00285442      M  ...                NaN                 NaN     7969

[5 rows x 12 columns]

**0).** **Сколько всего возрастных категорий?**  

In [7]:
data.columns

Index(['User_ID', 'Product_ID', 'Gender', 'Age', 'Occupation', 'City_Category',
       'Stay_In_Current_City_Years', 'Marital_Status', 'Product_Category_1',
       'Product_Category_2', 'Product_Category_3', 'Purchase'],
      dtype='object')

In [8]:
len(data['Age'].unique())

7

**Ответ: 7 возрастных категорий.**


**1).** **Сколько строк с мужчинами из категории города A? (речь не об уникальных ID мужчин, а о количестве строк)**




In [9]:
len(data[(data['City_Category']=='A') & (data['Gender']=='M')])

109831

**Ответ: 109831  строк с мужчинами из категории города A.**

**2).** **Сколько женщин от 46 до 50, потративших (столбец Purchase) больше 20000 (условных единиц, в данном случае)?   (речь не об уникальных ID, а о количестве строк)** 


In [10]:
len(data[(data['Age']=='46-50') & (data['Gender']=='F') & (data['Purchase'] > 20000)])

270

**Ответ: 270.**

**3).** **Сколько NaN'ов в столбце Product_Category_3?**

In [11]:
sum(data['Product_Category_3'].isna())

373299

**Ответ: 373299.**

**4).** **Какую долю (вещественное число от 0 до 1, округлить до 4-го знака) от всех покупателей составляют ВМЕСТЕ мужчины от 26 до 35 лет и женщины старше 36 лет (то есть нужно учесть несколько возрастных категорий)? (речь не об уникальных ID, а о количестве таких строк)**

In [12]:
data.shape


(537577, 12)

In [14]:
data['Age'].unique()

array(['0-17', '55+', '26-35', '46-50', '51-55', '36-45', '18-25'],
      dtype=object)

In [13]:
M = len(data[(data['Age']=='26-35') & (data['Gender']=='M')])
F = len(data[(data['Age'].isin(['55+', '46-50', '51-55', '36-45'])) & (data['Gender']=='F')])
(M + F) / 537577

0.4077201963625676

**Ответ: 0.4077.**